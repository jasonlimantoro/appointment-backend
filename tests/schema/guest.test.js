import gql from 'graphql-tag';
import { createTestClientAndServer } from '../utils';
import mockedGuests from '../../fixtures/guests';
import mockedEntries from '../../fixtures/entries';

describe('Guest schema', () => {
  it('listGuest: should work', async () => {
    const { query, guestAPI, entryAPI } = createTestClientAndServer();
    guestAPI.list = jest.fn(() => mockedGuests);
    entryAPI.byGuestId = jest.fn().mockResolvedValue(mockedEntries);
    const LIST_GUEST = gql`
      query guests {
        listGuest {
          id
          firstName
          lastName
          email
          entryToday {
            id
            see
            createdAt
          }
        }
      }
    `;
    const result = await query({ query: LIST_GUEST });
    expect(result).toMatchSnapshot();
    expect(guestAPI.list).toBeCalled();
    expect(entryAPI.byGuestId).toBeCalled();
  });

  it('getGuest: should work', async () => {
    const { query, guestAPI } = createTestClientAndServer();
    guestAPI.get = jest.fn(() => mockedGuests[0]);
    const GET_GUEST = gql`
      query guest($id: String!) {
        getGuest(id: $id) {
          id
          firstName
          lastName
          email
        }
      }
    `;
    const result = await query({
      query: GET_GUEST,
      variables: { id: 'some-id' },
    });
    expect(guestAPI.get).toBeCalledWith('some-id');
    expect(result).toMatchSnapshot();
  });

  it('byName: should work', async () => {
    const { query, guestAPI } = createTestClientAndServer();
    const guest = mockedGuests[0];
    const { firstName, lastName } = guest;
    guestAPI.byName = jest.fn().mockResolvedValue(guest);
    const BY_NAME = gql`
      query guestByName($input: ByNameInput!) {
        byName(input: $input) {
          id
          firstName
          lastName
          email
        }
      }
    `;
    const result = await query({
      query: BY_NAME,
      variables: { input: { firstName, lastName } },
    });
    expect(guestAPI.byName).toBeCalledWith({ firstName, lastName });
    expect(result).toMatchSnapshot();
  });

  it('create guest: should work', async () => {
    const { mutate, guestAPI } = createTestClientAndServer();
    const attributes = {
      firstName: 'John',
      lastName: 'Doe',
      email: 'john@example.com',
      NIK: '12345',
    };
    guestAPI.create = jest
      .fn()
      .mockResolvedValue({ ...attributes, id: 'some-id' });

    const CREATE_GUEST = gql`
      mutation CreateGuest($input: CreateGuestInput!) {
        createGuest(input: $input) {
          id
          firstName
          lastName
          email
          NIK
          company
        }
      }
    `;
    const result = await mutate({
      mutation: CREATE_GUEST,
      variables: { input: attributes },
    });
    expect(guestAPI.create).toBeCalledWith(attributes);
    expect(result).toMatchSnapshot();
  });
});
