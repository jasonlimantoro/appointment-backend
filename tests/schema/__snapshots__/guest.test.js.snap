// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest schema byName: should work 1`] = `
Object {
  "data": Object {
    "byName": Object {
      "NIK": "4532711991301150",
      "email": "Dejon.Beier@Huel.name",
      "firstName": "Maurine",
      "lastName": "Rau",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Guest schema create guest: should work 1`] = `
Object {
  "data": Object {
    "createGuest": Object {
      "NIK": "12345",
      "company": null,
      "email": "john@example.com",
      "firstName": "John",
      "lastName": "Doe",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Guest schema getGuest: should work 1`] = `
Object {
  "data": Object {
    "getGuest": Object {
      "NIK": "4532711991301150",
      "email": "Dejon.Beier@Huel.name",
      "firstName": "Maurine",
      "lastName": "Rau",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Guest schema listGuest: should work 1`] = `
Object {
  "data": Object {
    "listGuest": Array [
      Object {
        "NIK": "4532711991301150",
        "email": "Dejon.Beier@Huel.name",
        "entryToday": Array [],
        "firstName": "Maurine",
        "lastName": "Rau",
      },
      Object {
        "NIK": "4929542228489240",
        "email": "Darrion.Greenfelder@Mueller.org",
        "entryToday": Array [],
        "firstName": "Myrna",
        "lastName": "Mueller",
      },
      Object {
        "NIK": "5114446720603342",
        "email": "Marjorie.Stroman@Friesen.com",
        "entryToday": Array [],
        "firstName": "Audrey",
        "lastName": "Satterfield",
      },
      Object {
        "NIK": "6011973211683353",
        "email": "Gleason_Johnathon@yahoo.com",
        "entryToday": Array [],
        "firstName": "Jayme",
        "lastName": "Kuhlman",
      },
      Object {
        "NIK": "5451502535414044",
        "email": "Zander_Auer@yahoo.com",
        "entryToday": Array [],
        "firstName": "Marcelle",
        "lastName": "Gleason",
      },
      Object {
        "NIK": "5391346992204893",
        "email": "Genoveva.Huels@Quitzon.name",
        "entryToday": Array [],
        "firstName": "Laurence",
        "lastName": "Haag",
      },
      Object {
        "NIK": "4024007138332513",
        "email": "Katrina_Braun@Brenda.tv",
        "entryToday": Array [],
        "firstName": "Zechariah",
        "lastName": "Abshire",
      },
      Object {
        "NIK": "5274384958794918",
        "email": "Else_Abernathy@hotmail.com",
        "entryToday": Array [],
        "firstName": "Enid",
        "lastName": "Klein",
      },
      Object {
        "NIK": "348129022643854",
        "email": "Layne.Yost@Hegmann.me",
        "entryToday": Array [],
        "firstName": "Yadira",
        "lastName": "Oberbrunner",
      },
      Object {
        "NIK": "4532557118938778",
        "email": "Amanda_Jenkins@yahoo.com",
        "entryToday": Array [],
        "firstName": "Reyes",
        "lastName": "Block",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
